axelor-talent/src/main/java/com/axelor/apps/talent/db/repo/JobApplicationTalentRepository.java

```diff
@@ -39,4 +41,18 @@ public JobApplication save(JobApplication entity) {
 
     return super.save(entity);
   }
+
+  @Override
+  public void remove(JobApplication entity) {
+
+    super.remove(entity);
+
+    Beans.get(EventRepository.class)
+        .all()
+        .filter(
+            "relatedToSelect = ?1 AND self.relatedToSelectId = ?2",
+            JobApplication.class.getName(),
+            entity.getId())
+        .remove();
+  }
```

Comment:
Might be better to use a entity listener with the PreRemove annotation to not forget any case.  Just be careful to use the autoflush at false when using repository in listener.